src/binary/object.cpp

```diff
@@ -16,6 +16,7 @@
   #include <link.h> // needed for dladdr1's link_map info
  #endif
 #elif IS_WINDOWS
+ #define WIN32_LEAN_AND_MEAN
```

Comment:
Do you think it'd be worthwhile to just do `target_compile_definitions(... PRIVATE WIN32_LEAN_AND_MEAN)`?